frontend/packages/knative-plugin/src/components/traffic-splitting/TrafficSplittingModal.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { FormikProps, FormikValues } from 'formik';
+import {
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+} from '@console/internal/components/factory/modal';
+import { TextInputTypes } from '@patternfly/react-core';
+import {
+  MultiColumnField,
+  InputField,
+  DropdownField,
+} from '@console/dev-console/src/components/formik-fields';
+
+export interface TrafficSplittingModalProps {
+  revisionItems: any;
+  errorMessage?: string;
+  onCancel: () => void;
+}
+
+type Props = FormikProps<FormikValues> & TrafficSplittingModalProps;
+
+const TrafficSplittingModal: React.FC<Props> = ({
+  errorMessage,
+  revisionItems,
+  handleSubmit,
+  isSubmitting,
+  onCancel,
+}) => {
+  const handleSave = (e) => {
+    e.preventDefault();
+    handleSubmit();
+  };
+  return (
+    <form className="modal-content" onSubmit={handleSave}>
+      <ModalTitle>Set traffic distrubution</ModalTitle>
+      <ModalBody>
+        <p>Set traffic distribution for the Revisions of the Knative Service</p>
+        <MultiColumnField
+          name="trafficSplitting"
+          addLabel="Add Revision"
+          headers={['Split', 'Tag', 'Revision']}
+          emptyValues={{ percent: '', tag: '', revisionName: '' }}
+        >
+          <InputField name="percent" type={TextInputTypes.number} placeholder="100" required />
```

Comment:
Since, its a `number` type input this will not cover full width of the column because of a CSS override from console. You can use `style={{ maxWidth: '100%' }}` if you want this to be full width. Ignore if not needed based on UX.